R/data_color.R

```diff
@@ -439,6 +443,70 @@ get_alpha_vec <- function(colors) {
   )
 }
 
+#' Get the rgba color representation of a color
+#'
+#' @noRd
+get_rgba_vec <- function(colors) {
+
+  build_rgba_string <- function(color, alpha_val) {
+
+    grDevices::col2rgb(color)[, 1] %>%
+      unname() %>%
+      append(alpha_val) %>%
+      paste(collapse = ",") %>%
+      paste_left("rgba(") %>%
+      paste_right(")")
+  }
+
+  vapply(
+    colors,
+    FUN.VALUE = character(1),
+    USE.NAMES = FALSE,
+    function(color) {
+
+      if (color %>% is_hex_col_no_alpha()) {
+
+        build_rgba_string(color = color, alpha_val = 1)
+
+      } else if (color %>% is_hex_col_w_alpha()) {
+
+        alpha_dec <-
+          (
+            color %>%
+              substring(8, 9) %>%
+              paste_left("0x") %>%
+              strtoi()
+          ) / 256
+
+        build_rgba_string(color = color, alpha_val = alpha_dec)
+
+      } else if (tolower(color) %in% grDevices::colors()) {
+
+        build_rgba_string(color = tolower(color), alpha_val = 1)
+
+      } else {
+        NA_character_
+      }
+    })
+}
+
+rgba2hex <- function(rgba_col) {
+
+  channels <-
+    rgba_col %>%
+    tidy_gsub(pattern = "(rgba\\(|\\))", "") %>%
+    strsplit(",") %>%
+    unlist() %>%
+    as.numeric()
+
+  hex_color <-
+    grDevices::rgb(
+      red = channels[1]/255,
+      green = channels[2]/255,
+      blue = channels[3]/255
+    )
+}
```

Comment:
This doesn't return anything. I think I have to fix this as a first commit before the internal documentation.

Reply:
Fixed in the renamed `rgba_to_hex()` function.